spec/facility_spec.rb

```diff
@@ -2,25 +2,284 @@
+                      engine: :ice} )  
+
+    @registrant_1 = Registrant.new('Bruce', 18, true )
+    @registrant_2 = Registrant.new('Penny', 16 )
+    @registrant_3 = Registrant.new('Tucker', 15 )
   end
   describe '#initialize' do
     it 'can initialize' do
-      expect(@facility).to be_an_instance_of(Facility)
-      expect(@facility.name).to eq('Albany DMV Office')
-      expect(@facility.address).to eq('2242 Santiam Hwy SE Albany OR 97321')
-      expect(@facility.phone).to eq('541-967-2014')
-      expect(@facility.services).to eq([])
+
+      expect(@facility_1).to be_an_instance_of(Facility)
+      expect(@facility_1.name).to eq('Albany DMV Office')
+      expect(@facility_1.address).to eq('2242 Santiam Hwy SE Albany OR 97321')
+      expect(@facility_1.phone).to eq('541-967-2014')
+      expect(@facility_1.services).to eq([])
     end
   end
 
-  describe '#add service' do
+  describe '#add_service' do
     it 'can add available services' do
-      expect(@facility.services).to eq([])
-      @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
-      @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+
+      expect(@facility_1.services).to eq([])
+
+      @facility_1.add_service('New Drivers License')
+      @facility_1.add_service('Renew Drivers License')
+      @facility_1.add_service('Vehicle Registration')
+
+      expect(@facility_1.services).to eq(['New Drivers License', 
+                                          'Renew Drivers License', 
+                                          'Vehicle Registration'])
+    end
+  end
+
+  describe "registering a vehicle" do
+    describe "register_vehicle" do
+      it "should register vehicles" do
+        @facility_1.add_service('Vehicle Registration')
+
+        expect(@facility_1.registered_vehicles).to eq([])
+        
+        @facility_1.register_vehicle(@cruz)
+
+        expect(@facility_1.registered_vehicles).to eq([@cruz])
+      end
+    end
+
+    describe "plate_type" do
+      it "should set a plate type" do
+        @facility_1.add_service('Vehicle Registration')
+
+        expect(@cruz.plate_type).to eq(nil)
+
+        @facility_1.register_vehicle(@cruz)
+        expect(@cruz.plate_type).to eq(:regular)
+      end
+    end
+
+    describe "fees" do
+      it "collects fees" do
+        @facility_1.add_service('Vehicle Registration')
+
+        expect(@facility_1.collected_fees).to eq(0)
+
+        @facility_1.register_vehicle(@cruz)
+
+        expect(@facility_1.collected_fees).to eq(100)
+
+      end
+    end
+
+    it "can register other vehicles" do
+      @facility_1.add_service('Vehicle Registration')
+      
+      @facility_1.register_vehicle(@cruz)
+
+      @facility_1.register_vehicle(@camaro)
+
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+
```

Comment:
Is there a reason that you don't have expect statements for `@cruz.registration_date` and `@cruz.plate_type`?

Reply:
It was probably an oversight on my part.